PyInstaller/utils/win32/winresource.py

```diff
@@ -202,7 +202,7 @@ def UpdateResources(dstpath, data, type_, names=None, languages=None):
             for language in res[type_][name]:
                 logger.info("Updating resource type %s name %s language %s",
                             type_, name, language)
-                win32api.UpdateResource(hdst, type_, name, data, language)
+                win32api.UpdateResource(hdst, type_, name, bytes(data, 'UTF-8'), language)
```

Comment:
I just tried out the python3 branch and can confirm this patch is needed. Encoding as UTF-8 seems to work - the program loader is apparently able to read and parse the manifest resource but I'll bet it's parsing it using an ANSI codepage instead of UTF-8. OTOH, it contains XML with an `encoding` attribute so it could very well switch to UTF-8. The only way to know for sure is to give it an assembly with foreign chars in its filename and see what it tries to load. I don't feel like going out of my way to do that. ;)

This change is still needed to fix the `TypeError: 'str' does not support the buffer interface` though. It's saying it wants a `bytes` and not a `str`.